File: schemas/persona.py
from utils.ma import ma
from model.persona import Persona
from marshmallow import fields
from schemas.usuario import UsuarioSchema
from schemas.ubicacion import UbicacionSchema

class PersonaSchema(ma.Schema):
    class Meta:
        model = Persona
        fields = (
            'id_persona',
            'ubicacion',
            'usuario', 
            'nombres',
            'apellidos',
            'fecha_nacimiento',
            'sexo',
            'estado_civil',
            'celular',
            'tipo_persona'
        )
    usuario = fields.Nested(UsuarioSchema)
    ubicacion = fields.Nested(UbicacionSchema)

persona_schema = PersonaSchema()
personas_schema = PersonaSchema(many=True)

File: v2-microservicios/sisvita-app/auth-server/src/main/java/sm/dsw/sisvita/auth/model/Persona.java
package sm.dsw.sisvita.auth.model;
import jakarta.persistence.*;
import lombok.*;
import java.util.Date;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "persona")
public class Persona {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_persona;

    @OneToOne
    @JoinColumn(name = "id_usuario", referencedColumnName = "id_usuario")
    private Usuario usuario;

    @Column(name = "nombres")
    private String nombres;

    @Column(name = "apellidos")
    private String apellidos;

    @ManyToOne
    @JoinColumn(name = "ubigeo", referencedColumnName = "ubigeo")
    private Ubicacion ubicacion;

    @Column(name = "fecha_nacimiento")
    private Date fecha_nacimiento;

    @Column(name = "sexo")
    private String sexo;

    @Column(name = "estado_civil")
    private String estado_civil;

    @Column(name = "celular")
    private String celular;

    @Column(name = "tipo_persona")
    private String tipo_persona;

}
